openshot-qt: add new package, version 2.0.7

tohl opened the pull request:
###### Motivation for this change

add free video editor openshot-qt and dependencies required for build
###### Things done
- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

FRidh commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -0,0 +1,48 @@
+{stdenv, fetchurl, fetchFromGitHub, callPackage, makeWrapper, doxygen
+, ffmpeg, python3, python3Packages, qt55}:
```
python3 comes with python3Packages, so please drop python3 here.

FRidh commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -0,0 +1,48 @@
+with stdenv.lib;
+
+let
+  libopenshot = callPackage ./libopenshot.nix {};
```
if you need `python` here, I suggest you use the let expression here for it.
```
let 
  python = `python3Packages.python`;
in ...

FRidh commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -0,0 +1,48 @@
+    cp -r src/* $(toPythonPath $out)
+    mkdir -p $out/bin
+    echo "#/usr/bin/env sh" >$out/bin/openshot-qt
+    echo "exec ${python3}/bin/python3 $(toPythonPath $out)/launch.py" >>$out/bin/openshot-qt
```
`${python3}/bin/python3` -> `${python.interpreter}`

FRidh commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -0,0 +1,48 @@
+    echo "exec ${python3}/bin/python3 $(toPythonPath $out)/launch.py" >>$out/bin/openshot-qt
+    chmod +x $out/bin/openshot-qt
+    wrapProgram $out/bin/openshot-qt \
+      --prefix PYTHONPATH : "$(toPythonPath $out):$(toPythonPath ${libopenshot}):$(toPythonPath ${python3Packages.pyqt5}):$(toPythonPath ${python3Packages.sip_4_16}):$(toPythonPath ${python3Packages.httplib2}):$PYTHONPATH"
```
you can use here `stdenv.lib.makeLibraryPath`

FRidh commented:
Thanks for the PR. Apologies for reviewing it this late. 
Could you put `unittest-cpp` in a separate commit and name the commits according to the contributing guidelines.

FRidh commented:
oh and cc me when that's done. It looks like it is to be ready to merged then.

Commit:
unittest-cpp: init at 1.6.1

Commit:
openshot-qt: init at 2.0.7

tohl commented:
Hi Frederik,

> python3 comes with python3Packages, so please drop python3 here.

fixed

> ${python3}/bin/python3 -> ${python.interpreter}

fixed

> you can use here stdenv.lib.makeLibraryPath

I tried and did not figure it out.  Also I haven't found any example of
doing this with PYTHONPATH, it is always without makeLibraryPath.

> Thanks for the PR. Apologies for reviewing it this late.

I thank you for your time and help!

> Could you put unittest-cpp in a separate commit

fixed

> and name the commits according to the contributing guidelines.

fixed

> oh and cc me when that's done. It looks like it is to be ready to
> merged then.

I implemented your suggestions, built and tested openshot-qt and it
worked fine.  Then I rebased on top of master and after that I get the
following error:

-- Detecting CXX compile features
Error: Failed to determine the detail of your Qt installation. Try again
using
the --verbose flag to see more detail about the problem.
Querying qmake about your Qt installation...
Determining the details of your Qt installation...
/nix/store/517s8rgjl7yda8s3kfzlxnp0zapsix13-qtbase-5.5.1-dev/bin/qmake
-o qtdetail.mk qtdetail.pro
Could not find qmake configuration file linux-g++.
Error processing project file: qtdetail.pro
builder for
‘/nix/store/dqbdd8fl5dhc6jshfzvz3hq4xfm39ixr-python3.5-PyQt-5.5.1.drv’
failed with exit code 1
cannot build derivation
‘/nix/store/fsqnnbhwg4p2f67ijampxqzw7hcd11w0-openshot-qt-2.0.7.drv’: 1
dependencies couldn't be built
error: build of
‘/nix/store/fsqnnbhwg4p2f67ijampxqzw7hcd11w0-openshot-qt-2.0.7.drv’
failed

This looks like a problem with the HEAD in python3.5-PyQt-5.5.1.  I have
no idea what to do about this and I think this is out of scope for the
openshot-qt nix package.

tohl commented:
cc @FRidh

FRidh commented:
> I tried and did not figure it out.  Also I haven't found any example of
doing this with PYTHONPATH, it is always without makeLibraryPath.

You're correct. I thought it could be used also for this case but apparently it can't.

Testing now.

FRidh commented:
Indeed, there is a issue with PyQt5. See https://github.com/NixOS/nixpkgs/issues/17045

tohl commented:
> Merged #16098.

Thanks a lot!
